Automattic/WP-Job-Manager
# Add ability to disable state storage locally or globally

Fixes #1812 

#### Changes proposed in this Pull Request:

* Cleanup: Remove method prefixes. Not necessary in scope.
* Adds new filter so site admins or plugins can add data attributes to wrapper for `[jobs]` shortcode.
* Adds ability to disable the form/results state storage on `[jobs]` both locally with data attribute or globally with class on `<body>`. This could be useful for cookie consent customization or if the theme provides an alternative approach.

#### Testing instructions:

Test both of these snippets individually and make sure when activated the `[jobs]` filter form doesn't store anything in local session and when you click onto a job listing result and then click back nothing is preserved (some browsers might automatically preserve the form state).

* Test with this snippet and make sure state isn't saved:
```php
add_filter( 'job_manager_jobs_shortcode_data_attributes', function( $data ) {
  $data['disable-form-state-storage'] = '1';
  
  return $data;
} );
```

* Test with this snippet and make sure state isn't saved:
```php
add_filter( 'body_class', function( $classes ) {
  $classes[] = 'disable-job-manager-form-state-storage';
  
  return $classes;
} );
```
* Test with both snippets removed. Click on job listing in `[jobs]` results and then click back. Make sure form and results are preserved and the results aren't reloaded.


#### Proposed changelog entry for your changes:
